### ISSUE TITLE
Stable Rust now that GATs are stable?

### ISSUE THREAD

**Issue by rtfeldman** [monoio-rs/monoio#129]
Thanks for making this library!

I noticed in the readme that GATs are a reason for depending on nightly Rust instead of stable. Now that GATs have just landed in stable, is it possible for monoio to depend on stable?

**Comment by ihciah** [comment]
Except for GAT, we rely on `type_alias_impl_trait` feature, and it's hard to get rid of it...

**Comment by jklop123** [comment]
With [async_fn_in_trait, return_position_impl_trait_in_trait](https://github.com/rust-lang/rust/issues/91611) has merged into the stable version (1.75), perhaps a non-GAT implementation based on the stable version will become possible.

**Comment by ihciah** [comment]
> With [async_fn_in_trait, return_position_impl_trait_in_trait](https://github.com/rust-lang/rust/issues/91611) has merged into the stable version (1.75), perhaps a non-GAT implementation based on the stable version will become possible.

Yes it is. It would become simpler to write trait and impl, but it may causes break changes. Also I don't know if there would be overhead when replace code like this:

```
type RetFut = impl Future<..>;
fn demo(&self, num: i32) -> Self::RetFut {...}
fn demo2(&self, num: i32) -> Self::RetFut {
    let num = nun + 1;
    self.demo(num)
}
```

to

```
async fn demo(&self, num: i32) -> xxx {...}
async fn demo2(&self, num: i32) -> xxx {
    let num = nun + 1;
    self.demo(num).await
}
```

If no overhead, I think it will be better to use the new style.
